### DISCUSSION TITLE
substring fragmentation algorithm

### DISCUSSION THREAD

**Discussion by xdxxxdx** [Informatievlaanderen/VSDS-LDESServer4J#767] [Q&A] [answered]
Hello,

I have some difficulties in understanding the substring fragmentation algorithm.
e.g. I have a company's name is: "ACKERMANS & VAN HAAREN" ;
The substring fragmentation property is set to the company name.
Root fragment is full.
The bucketizer 'A' is not full yet, so the member should be ONLY added under http://localhost:8080/kbo/by-name?substring=a right?

Cause what I got how is the member is added under:
![image](https://github.com/Informatievlaanderen/VSDS-LDESServer4J/assets/15314807/aa24ffcf-9ccb-4af3-9be1-6566bc5035b1)
Actually, from what I understand from the https://github.com/Informatievlaanderen/VSDS-LDESServer4J/blob/main/ldes-fragmentisers/ldes-fragmentisers-substring/README.MD
Each member should be only added one time to the fragmented stream right?

Thanks

**Comment by Yalz** [comment, marked as answer]
Hi @xdxxxdx , 
I do believe this is intended behavior. It considers every word as a separate part to do the fragmentation on.
Since your name is "ACKERMANS & VAN HAAREN", it will put it in a bucket of "A", "&", "V" and "H".

**Reply by Yalz** [reply]
@WLefever-Cegeka @ddvlanck feel free to correct me

**Reply by Tomvbe** [reply]
This is correct, each member is **added one time per token**. Looking back at the readme I agree that the example can cause some confusion. This is because the use of tokens was added later to the algorithm. I will make a PR to improve the example description.

**Reply by xdxxxdx** [reply]
Thanks a lot for the reply.

**Reply by Tomvbe** [reply]
cfr. https://github.com/Informatievlaanderen/VSDS-LDESServer4J/pull/781
